mne/event.py

```diff
@@ -885,3 +886,349 @@ def concatenate_events(events, first_samps, last_samps):
+            the interval is between "a (s)" and "b (s)".
+            If a is None the beginning of the data is used
+            and if b is None then b is set to the end of the interval.
+            If baseline is equal to (None, None) all the time
+            interval is used.
+            The baseline (a, b) includes both endpoints, i.e. all
+            timepoints t such that a <= t <= b.
+        detrend : int | None
+            If 0 or 1, the data channels (MEG and EEG) will be detrended when
+            loaded. 0 is a constant (DC) detrend, 1 is a linear detrend. None
+            is no detrending. Note that detrending is performed before baseline
+            correction. If no DC offset is preferred (zeroth order detrending),
+            either turn off baseline correction, as this may introduce a DC
+            shift, or set baseline correction to use the entire time interval
+            (will yield equivalent results but be slower).
+        stim_channel : None | string | list of string
+            Name of the stim channel or all the stim channels
+            affected by the trigger. If None, the config variables
+            'MNE_STIM_CHANNEL', 'MNE_STIM_CHANNEL_1', 'MNE_STIM_CHANNEL_2',
+            etc. are read. If these are not found, it will fall back to
+            'STI 014' if present, then fall back to the first channel of type
+            'stim', if present.
+        mask : int | None
+            The value of the digital mask to apply to the stim channel values.
+            If None (default), no masking is performed.
+        uint_cast : bool
+            If True (default False), do a cast to ``uint16`` on the channel
+            data. This can be used to fix a bug with STI101 and STI014 in
+            Neuromag acquisition setups that use channel STI016 (channel 16
+            turns data into e.g. -32768), similar to ``mne_fix_stim14 --32``
+            in MNE-C.
+        mask_type: 'and' | 'not_and'
+            The type of operation between the mask and the trigger.
+            Choose 'and' for MNE-C masking behavior.
+        """
+
+        from .epochs import Epochs
+        if isinstance(category, str):
+            category = self[category]
+        if reject is True:
+            reject = self.reject
+        if flat is True:
+            flat = self.flat
+        mne_events = find_events(raw, stim_channel=stim_channel,
+                                 mask=mask, mask_type=mask_type, output='step',
+                                 uint_cast=uint_cast, consecutive=True,
+                                 verbose=False)
+        sfreq = raw.info['sfreq']
+        # create array of category reference times (t0),
+        # and corresponding (fake) event_id for mne.Epochs
+        cat_t = self._mne_events_to_category_t0(category, mne_events, sfreq)
+        catev = np.c_[cat_t, np.zeros(cat_t.shape),
+                      np.ones(cat_t.shape)].astype(np.uint32)
+        id = {category['comment']: 1}
```

Comment:
(I added a space when reverting the change, so GitHub was hiding these line notes. Trying to keep it visible for discussion)